code/modules/hippie/recycling/trapdoor.dm

```diff
@@ -0,0 +1,171 @@
+//Trap Door
+/obj/machinery/disposal/trapdoor
+	name = "trapdoor"
+	desc = "Almost like a door, but on floor."
+	density = FALSE
+	icon = 'icons/hippie/obj/atmospherics/trapdoor.dmi'
+	icon_state = "closed"
+	mode = FALSE
+	layer = 2.01
+	var/id = 1
+	var/auto_close = 1200
+	var/auto_close_on_mob = 100
+	var/sound_open = 'sound/hippie/machines/blast_door.ogg'
+	var/sound_close = 'sound/hippie/machines/blast_door.ogg'
+	var/open = FALSE
+
+/obj/machinery/disposal/trapdoor/proc/open()
+	if(flushing)
+		return
+	if(open)
+		return
+	open = TRUE
+	flick("opening", src)
+	playsound(loc, sound_open, 100, 1)
+	icon_state = "open"
+	spawn(5)
+		for(var/mob/living/M in loc)
+			if(!M.floating)
+				M.forceMove(src)
+				trap_flush()
+				if(auto_close_on_mob)
+					spawn(auto_close_on_mob)
+						close()
+		for(var/obj/item/O in loc)
+			if(!O.throwing || !O.anchored)
+				O.forceMove(src)
+				trap_flush()
+	if(auto_close)
+		spawn(auto_close)
+			close()
+	return 1
+
+
+/obj/machinery/disposal/trapdoor/proc/close()
+	if(open)
+		flick("closing", src)
+		icon_state = "closed"
+		playsound(loc, sound_open, 100, 1)
+		open = FALSE
+		return 1
+
+/obj/machinery/disposal/trapdoor/New(loc,var/obj/structure/disposalconstruct/make_from)
+	..()
+	stored.ptype = DISP_END_CHUTE
+	spawn(5)
```

Comment:
Please use the `addtimer` and `callback` system to do this instead of using spawns, you can find more information about this [here](https://github.com/HippieStation/HippieStation13/blob/master/code/datums/callback.dm)